### ISSUE TITLE
linuxfib: deal with blackhole/unreachable routes

### ISSUE THREAD

**Issue by israel-lugo** [israel-lugo/netforeman#5]
We get a ``NetlinkError(22, "Invalid argument")`` when trying to do a ``LinuxFIBInterface.get_route_to()`` and the route happens to be a blackhole. Deal with this somehow.

iproute2 has the same behavior, seems to be netlink-related. However, iproute2 has the ``ip route list match`` operator, which can show us routes that match even if they're blackholed. See how they do that. Can't use ``IPRoute.get_routes(dst=)``, that fails with a NetlinkError too.

**Comment by israel-lugo** [comment]
It works if we use ``Route.route("dump", dst=)``. This doesn't resolve routes, it just lists them. The error seems to be generated by the lookup machinery, so listing doesn't trigger it. However, it's not practical to find a matching route this way. It only works for exact matches.

We could dump everything, and do the filtering ourselves, using ``IPRoute.get_routes(match=)``:
```python
>>> f.ipr.get_routes(match=lambda x: x.get_attr('RTA_DST') is not None and netaddr.IPNetwork(x.get_attr('RTA_DST')) in netaddr.IPNetwork('198.19.66.0/24') )
[{'scope': 0, 'event': 'RTM_NEWROUTE', 'type': 6, 'src_len': 0, 'header': {'sequence_number': 307, 'length': 44, 'error': None, 'type': 24, 'flags': 2, 'pid': 3561}, 'tos': 0, 'proto': 3, 'attrs': [('RTA_TABLE', 254), ('RTA_DST', '198.19.66.0')], 'table': 254, 'flags': 0, 'family': 2, 'dst_len': 24}]
```

**Comment by israel-lugo** [comment]
iproute2 is doing the matching in userland. See ``ip/iproute.c:filter_nlmsg()``, ``filter.mdst`` (for the "match" selector).

**Comment by israel-lugo** [comment]
As it turns out, we fixed this wrong in 15738bd. Gentoo kernel 4.4.26 gave ENETUNREACH in testing, but the documented behavior is to return one of (EHOSTUNREACH, EACCES, EINVAL), which is what Debian kernel 4.6.0 does. See man ip-route(8), linux/net/ipv4/fib_semantics.c:fib_props and linux/net/ipv6/fib6_rules.c:fib6_rule_action().
